### ISSUE TITLE
Windows: flashing terminal windows when invoking shell

### ISSUE THREAD

**Issue by sewbacca** [love2d/love#1995]
When calling `io.popen` or `os.execute` on Windows, terminals start flashing on and off the screen, when no console window is open.
I would've expected that these commands would be executed silently in the background.

**Reproduction steps**

- Windows 10
- love2d 11.5

```lua
local last = 0
function love.update()
	-- Using a timer, such that it's not that aggressive
	if love.timer.getTime() - last > 1 then
		os.execute("echo test")
		io.popen("echo test"):read "*a"
		last = love.timer.getTime()
	end
end
```

**Comment by slime73** [comment]
As far as I know this is just how Microsoft made things behave on Windows for windowed apps that don't already have a console, I don't think it's something love can change.

**Comment by akkartik** [comment]
I just tried this example on Windows with the `lovec.exe` that comes with LÖVE, and the window stops flashing and moving.

**Comment by sewbacca** [comment]
> I just tried this example on Windows with the `lovec.exe` that comes with LÖVE, and the window stops flashing and moving.

If a console is already open there is no issue. You can open a console with `love._openConsole()`, when called via `love.exe`. The issue only appears when no console window is open.

**Comment by slime73** [comment]
Yeah, Windows basically makes sure a console is open inside those system function calls. If there isn't one already then it creates a temporary one for the duration of the call.

**Comment by akkartik** [comment]
:bulb:

**Comment by sewbacca** [comment]
I've found a workarround, using `ffi`. One opens a console with `love._openConsole()` and then hides it using the WinAPI:
```lua
local ffi = require "ffi"

ffi.cdef [[
	void* __stdcall GetConsoleWindow();
	int __stdcall ShowWindow(void* hWnd, int nCmdShow);
]]

local function hideConsole()
	love._openConsole()
	local console = ffi.C.GetConsoleWindow()
	ffi.C.ShowWindow(console, 0)
end

hideConsole()
```
All future calls to `io.popen` or `os.execute` will not open new terminal windows, but write into the hidden window.
I also do not get any first time flashes, which I expected, so assuming `ffi` is available, this is basically a perfect fix.
